src/Store.ts

```diff
@@ -0,0 +1,582 @@
+	//		The object in the store that matches the given id.
+	get: (id: string | number) => any;
+
+	// idProperty: String
+	//		Indicates the property to use as the identity property. The values of this
+	//		property should be unique.
+	idProperty: string;
+
+	// Model: Function
+	//		This should be a entity (like a class/constructor) with a 'prototype' property that will be
+	//		used as the prototype for all objects returned from this store. One can set
+	//		this to the Model from dmodel/Model to return Model objects, or leave this
+	//		to null if you don't want any methods to decorate the returned
+	//		objects (this can improve performance by avoiding prototype setting),
+	Model: NewableStoreModel;
+
+	// parse: Function
+	//		One can provide a parsing function that will permit the parsing of the data. By
+	//		default we assume the provide data is a simple JavaScript array that requires
+	//		no parsing (subclass stores may provide their own default parse function)
+	parse: (...args: any[]) => any;
+
+	// summary:
+	//		Stores an object
+	// object: Object
+	//		The object to store.
+	// directives: dstore/Store.PutDirectives?
+	//		Additional directives for storing objects.
+	// returns: Object
+	//		The object that was stored, with any changes that were made by
+	//		the storage system (like generated id)
+	put: (object: any, directives?: dstore.PutDirectives) => any;
+
+	// queryAccessors: Boolean
+	//		Indicates if client-side query engine filtering should (if the store property is true)
+	//		access object properties through the get() function (enabling querying by
+	//		computed properties), or if it should (by setting this to false) use direct/raw
+	// 		property access (which may more closely follow database querying style).
+	queryAccessors: boolean =  true;
+
+	// summary:
+	//		Deletes an object by its identity
+	// id: Number
+	//		The identity to use to delete the object
+	remove: (id: any) => Promise<Object>;
+
+	// queryLog: __QueryLogEntry[]
+	//		The query operations represented by this collection
+	queryLog: dstore.QueryLogEntry<any>[];
+
+	storage: Evented;
+
+	// stringify: Function
+	//		For stores that serialize data (to send to a server, for example) the stringify
+	//		function can be specified to control how objects are serialized to strings
+	stringify: () => any;
+
+	constructor(options?: StoreArgs) {
+		super();
+		this.autoEmitEvents = true;
+		this.autoEmitHandles = [];
+		this.idProperty = 'id';
+		this.queryLog = [];
+		this.Filter = Filter;
+		// Set defaults
+		// perform the mixin
+		options && lang.mixin(this, options);
+		// TODO - Don't do this, it's a workaround to make inheritance from this class via
+		// declare work properly.
+		for (let name in this) {
+			let value = this[name];
+			if (Object.prototype.toString.call(value) === '[object Function]') {
+				value.nom = name;
+			}
+		}
+
+		// TODO - This is designed specifically for working with declare-based inheritance and needs to be reworked.
```

Comment:
I might be in favor of cutting all explicit ties to models from dstore (not making it incompatible with the idea, but rather just not trying to go out of its way to hook them up itself), but that's almost definitely something to look into later.